### PR TITLE
Fix react 18 animation flicker

### PR DESCRIPTION
**Pull request by snowystinger**
Closes https://github.com/adobe/react-spectrum/issues/4419

Notes in issue discussion

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## 📝 Test Instructions:

Should be no flash of content after new dialog story is closed

### CODE REVIEW

**Comment by snowystinger** [comment]
GET_BUILD

File: packages/react-aria-components/stories/animations.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  Heading,
+  Modal,
+  ModalOverlay
+} from 'react-aria-components';
+import React from 'react';
+import styles from './styles.css';
+
+
+export default {
+  title: 'React Aria Components - Animations'
```
Comment (ktabors) [inline packages/react-aria-components/stories/animations.stories.tsx:15]: This seems like a weird place for. Are we planning for more RAC animations? Shouldn't this be under RAC?

File: packages/react-aria-components/stories/animations.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+};
+
+export let ModalAnimation = {
```
Comment (ktabors) [inline packages/react-aria-components/stories/animations.stories.tsx:18]: I'm not able to see the flicker with the original story. It fades in slow and it fades out slow.
In this version:
- it fades in slow completely and I close it with the button, it jumps to 3/4 closed and fades out.
- it fades in slow completely and I close it with escape it either completely closes or 7/8 closes and fades out.
- If I press escape as it's fading it, it jumps to fully open and starts fading out.

File: packages/react-aria-components/stories/animations.stories.tsx (hunk 1 above)
Comment (snowystinger) [reply packages/react-aria-components/stories/animations.stories.tsx:15]: Yeah, wasn't too sure where to put it, seemed out of place with the RAC stories, seemed out of place with our other stories. I think we can decide where to move it later. Maybe other animation related bugs will come in or maybe we should have a story group called "Bugs"

File: packages/react-aria-components/stories/animations.stories.tsx (hunk 2 above)
Comment (snowystinger) [reply packages/react-aria-components/stories/animations.stories.tsx:18]: I don't understand the comment. But you must let the animation play all the way through, there's no logic in place to have it do things like reverse mid-way

**Review by LFDanLu — APPROVED** [review]

**Review by ktabors — APPROVED** [review]

**Pull request merged**